src/sentry/ownership/grammar.py

```diff
@@ -581,6 +581,26 @@ def resolve_actors(owners: Iterable[Owner], project_id: int) -> Mapping[Owner, A
     return {o: actors.get((o.type, o.identifier.lower())) for o in owners}
 
 
+def remove_deleted_owners_from_schema(
```

Comment:
Remove the deleted owners from the rule owners. If all rule owners have been deleted, remove the rule from the schema